### PR TITLE
PP-1030: Pbs_account fails to validate domained USER2 from logon domained USER1

### PR DESCRIPTION
**Pull request by bremanandjk**
#### Bug/feature Description
* *Pbs_account fails to validate domain USER2 from logon domain USER1*
* *While we try to install using domain user account and choosing different domain service account, it fails to validate due to the cause of determining domain controller name*
* *If there is an issue ID, link to it: [PP-1030](https://pbspro.atlassian.net/browse/PP-1030)*

#### Affected Platform(s)
*  *Windows*

#### Cause / Analysis / Design
* *Fails to fetch domain controller name*

#### Solution Description
* *Using windows api to fetch domain controller name before validating the user account*

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [x] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [ ] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/tools/pbs_account_win.c [hunk 1]
```diff
@@ -722,6 +724,17 @@ add_service_account(char *password)
 	mbstowcs(dnamew, dname, PBS_MAXHOSTNAME+1);
 	mbstowcs(dctrlw, dctrl, PBS_MAXHOSTNAME+1);
 
+	if (in_domain_environment && dctrlw[0] == '\0' ) {
+		if (NERR_Success == NetGetDCName(NULL, dnamew, (LPBYTE *)&dcw)) {
```
[the remark this answers is not in the document]
Comment (bremanandjk) [reply src/tools/pbs_account_win.c]: Thanks for reviewing @bhroam . 
neither LPWSTR nor LPBYTE is a structure, 
Here symbol LPCWSTR is defined as ```typedef const WCHAR* LPCWSTR;``` and 
LPBYTE is defined as ```typedef BYTE far            *LPBYTE;``` at c:\Program Files\Microsoft SDKs\Windows\v6.0A\Include\WinDef.h

File: src/tools/pbs_account_win.c [hunk 2]
```diff
@@ -722,6 +724,17 @@ add_service_account(char *password)
...
+			wcstombs(dctrl_buf, dcw, PBS_MAXHOSTNAME+1 );
+			mbstowcs(dctrlw, dctrl_buf, PBS_MAXHOSTNAME+1);
```
[the remark this answers is not in the document]
Comment (bremanandjk) [reply src/tools/pbs_account_win.c]: Done

**Comment by bremanandjk** [comment]
Thanks, @bhroam & @sid2364  for your approvals. 
@PBSPro/pbspro-maintainers , please take this PR further to merge.

**Comment by bhroam** [comment]
I merged in the PR

**Pull request merged**